dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceApplication.java

```diff
@@ -463,13 +468,16 @@ public void run(DockstoreWebserviceConfiguration configuration, Environment envi
         // optional CORS support
         // Enable CORS headers
         // final FilterRegistration.Dynamic cors = environment.servlets().addFilter("CORS", CrossOriginFilter.class);
-        final FilterHolder filterHolder = environment.getApplicationContext().addFilter(CrossOriginFilter.class, "/*", EnumSet.of(REQUEST));
+        CORS_ENDPOINTS.stream().forEach(urlContext -> {
+            FilterHolder filterHolder = environment.getApplicationContext().addFilter(CrossOriginFilter.class, urlContext, EnumSet.of(REQUEST));
+
+            filterHolder.setInitParameter(ACCESS_CONTROL_ALLOW_METHODS_HEADER, "GET,POST,DELETE,PUT,OPTIONS,PATCH");
+            filterHolder.setInitParameter(ALLOWED_ORIGINS_PARAM, "*");
+            filterHolder.setInitParameter(ALLOWED_METHODS_PARAM, "GET,POST,DELETE,PUT,OPTIONS,PATCH");
```

Comment:
Does `HEAD` need to be here, and on line 474? It wasn't before, but wondering if it should be.

Reply:
I'm tempted to say no because 1) the current launch-with pages work without a `HEAD` request (not sure why they'd need it). 2) Afaik all preflight requests should be handled with `OPTIONS`

Reply:
If there's no drawback to including `HEAD`, IMHO, might as well, could be useful to someone someday (for checking the existence of something or system status or whatever).